design/aws-ecs-service-discovery-integration.md

```diff
@@ -0,0 +1,235 @@
+   *
+   * @default PrivateDns
+   */
+  type?: cloudmap.NamespaceType;
+
+  /**
+   * The Amazon VPC that you want to associate the namespace with. Required for Private DNS namespaces
+   *
+   * @default VPC of the cluster for Private DNS Namespace, otherwise none
+   */
+  vpc?: ec2.IVpcNetwork;
+}
+```
+
+#### service#enableServiceDiscovery(options: ServiceDiscoveryOptions)
+
+This method would create a Cloud Map service, whose arn would then be passed as the serviceRegistry when the ECS service is created.
+
+Other fields in the service registry are optionally needed depending on the
+network mode of the task definition used by the ECS service.
+
+- If the task definition uses the bridge or host network mode, a containerName
+  and containerPort combination are needed. These will be taken from the
+defaultContainer on the task definition.
+
+- If the task definition uses the awsvpc network mode and a type SRV DNS record
+  is used, you must specify either a containerName and containerPort
+combination. These will be taken from the defaultContainer on the task definition.
+NOTE: In this case, the API also allows you to simply pass in "port" at the
+mutual exclusion of the `containerName` and `containerPort` combination, but
+for simplicity we are only including `containerName` and `containerPort` and
+not `port`.
+
+NOTE: warn about creating service with public namespace?
+
+If the customer wishes to have maximum configurability for their service, we will also add
+
+```ts
+export interface ServiceDiscoveryOptions {
+  /**
+   * Name of the cloudmap service to attach to the ECS Service
+   *
+   * @default CloudFormation-generated name
+   */
+  name?: string,
+
+  /**
+   * The DNS type of the record that you want AWS Cloud Map to create. Supported record types
+   * include A or SRV.
+
+   * @default: A
+   */
+  dnsRecordType?: cloudmap.DnsRecordType.A | cloudmap.DnsRecordType.SRV,
+
+  /**
+   * The amount of time, in seconds, that you want DNS resolvers to cache the settings for this
+   * record.
+   *
+   * @default 60
+   */
+  dnsTtlSec?: number;
+
+  /**
+   * The number of 30-second intervals that you want Cloud Map to wait after receiving an
+   * UpdateInstanceCustomHealthStatus request before it changes the health status of a service instance.
+   * NOTE: This is used for a Custom HealthCheckCustomConfig
+   */
+  failureThreshold?: number,
+}
+```
+
+A full example would look like the following:
+
+```
+const vpc = new ec2.VpcNetwork(stack, 'Vpc', { maxAZs: 2 });
+
+// Cloud Map Namespace
+const namespace = new servicediscovery.PrivateDnsNamespace(stack, 'MyNamespace', {
+  name: 'mydomain.com',
+  vpc,
+});
+
+// Cloud Map Service
+
+const cloudMapService = namespace.createService('MyCloudMapService', {
+  dnsRecordType: servicediscovery.DnsRecordType.A,
+  dnsTtlSec: 300,
+  customHealthCheck: {
+    failureThreshold = 1
+  }
+});
+
+// ECS Cluster
+const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
```

Comment:
You didn't call `cluster.addNamespace()` here, did you?

Reply:
oops.